device.go

```diff
@@ -41,8 +62,51 @@ func blockDeviceHandler(device pb.Device, spec *pb.Spec) error {
 		return nil
 	}
 
-	// At this point in the code, we assume the specification will be
-	// updated, meaning we should make sure we have valid pointers here.
+	return updateSpecDeviceList(device, spec)
+}
+
+func scsiBlockDeviceHandler(device pb.Device, mounts []*pb.Storage, spec *pb.Spec) error {
+	// Retrieve the device path from SCSI address.
+	devPath, err := getSCSIDevPath(device.Id)
+	if err != nil {
+		return err
+	}
+	device.VmPath = devPath
+
+	// If no ContainerPath is provided, this means we don't expect the
+	// device to be updated in the specification. Instead, this means
+	// this device is the rootfs, and the mount list should be updated
+	// accordingly.
```

Comment:
this document should be put in the API's documents too.

Reply:
Ok let me try to explain the situation here. In case of a rootfs, we need to both use `Device` and `Storage` because we need to detect the device being plugged inside the VM, and after it showed up, and that we have identified the name of the device, we need a `Storage` to mount the device to the expected rootfs path with the provided option such as `ext4` for instance.
This needs to be a 2 steps things IMO.

One way to avoid using empty `ContainerPath` (which I don't really like either), would be to add a new field `RootFs bool` to both structures like this:
```go
type Device struct {
        ...
        RootFs bool
        ...
}

...

type Storage struct {
        ...
        RootFs bool
        ...
}
```

WDYT ?

Reply:
> we need to both use Device and Storage because we need to detect the device being plugged inside the VM

using Storage only is enough, 'detecting the device being plugged inside the VM' should be done in the storage's driver.